### DISCUSSION TITLE
Compressed data size is larger than the original one after both compressed by gzip

### DISCUSSION THREAD

**Discussion by jz0224** [zeux/meshoptimizer#385] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/30064494/145179424-f09b0631-1ff4-46d6-881b-5f9bb62dfcd1.png)
![image](https://user-images.githubusercontent.com/30064494/145179571-aca26f5a-7000-4a64-a64b-412820800bb3.png)

Hi! The source data is my own customized mesh data. I found the data compressed by meshoptimizer is smaller than the original one, but after compressed by gzip, the compressed data becomes larger than the original one. Is this a common issue?

**Comment by zeux** [comment, marked as answer]
This can sometimes happen; no compression scheme can guarantee monotonicity. That said, it's atypical.

I would recommend checking if you observe the same result when using `gltfpack` (`-c` with or without gzip). Common issues that may prevent efficient compression are inefficient indexing, inefficient triangle order (for best results wrt compression size meshopt_optimizeVertexCacheStrip is recommended), and sometimes the use of floating point formats without filters can lead to complex compression size impact as well.

**Reply by jz0224** [reply]
Thank you for your reply! I converted my model to gltf and the meshopt compressed gltf model is smaller than the original one with gzip. So maybe there's some irrational data organization on my own mesh structure? I tried using meshopt_optimizeVertexCacheStrip before compress, the compressed file is smaller, but the gzip file is bigger.

**Comment by jz0224** [comment]
I followed your advice in https://github.com/zeux/meshoptimizer/discussions/383 (pad the normal data to 4 bytes), and the final data size (compressed by meshopt and gzip) is smaller than the original one (only compressed by gzip). Thanks for your great work!
